src/units/unit.cpp

```diff
@@ -2102,11 +2120,13 @@ void unit::apply_builtin_effect(std::string apply_to, const config& effect)
 			std::vector<std::string> temp_overlays = utils::parenthetical_split(add, ',');
 			std::vector<std::string>::iterator it;
 			for(it=temp_overlays.begin();it<temp_overlays.end();++it) {
-				overlays_.push_back(*it);
+				auto itr = std::find(obj_overlays_.begin(), obj_overlays_.end(), *it);
+				if (itr != obj_overlays_.end()){}
+				else {obj_overlays_.push_back(*it);}
```

Comment:
If you only need the else clause, please rewrite the conditional so that it doesn't need an else at all. However, as itr is only used for the comparison then I'd prefer it to be rewritten to not store itr in a named variable at all, instead just compare the result of find() to end().